pkg/marker/marker.go

```diff
@@ -0,0 +1,126 @@
+package marker
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	clientset kubernetes.Interface
+)
+
+const (
+	resourceNamespace = "ovs-cni.network.kubevirt.io"
+)
+
+type patchOperation struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value,omitempty"`
+}
+
+func init() {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		glog.Fatalf("Error while obtaining cluster config: %v", err)
+	}
+
+	clientset, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		glog.Fatalf("Error building example clientset: %v", err)
+	}
+}
+
+func getAvailableResources() (map[string]bool, error) {
+	outputRaw, err := exec.Command(
+		"ovs-vsctl",
+		"--db", "unix:///host/var/run/openvswitch/db.sock",
```

Comment:
So this is what makes us run the daemon privileged, right? I wonder if there is a way to access the socket with lowered privileges. I see the socket is available to openvswitch user only on my fedora host. I guess it means we either run the container as openvswitch user, or privileged indeed. Sigh. :(